### PR TITLE
Add nonce to SilentRequest

### PR DESCRIPTION
**Pull request by jo-arroyo**
Making `nonce` available and optional on SilentRequest.ts

### CODE REVIEW

**Review by konstantin-msft — APPROVED** [review]

File: lib/msal-browser/src/request/SilentRequest.ts [hunk 1]
```diff
@@ -47,4 +47,5 @@ export type SilentRequest = Omit<
     prompt?: string;
     state?: string;
     tokenBodyParameters?: StringDict;
+    nonce?: string;
```
Comment (tnorling) [inline lib/msal-browser/src/request/SilentRequest.ts:50]: What's the behavior when a token is served from cache?

Comment (hectormmg) [reply lib/msal-browser/src/request/SilentRequest.ts:50]: +1, should nonce be added as a lookup filter?

Comment (tnorling) [reply lib/msal-browser/src/request/SilentRequest.ts:50]: Might need to be treated as a force-iframe refresh as it only gets passed to the /authorize endpoint. @jo-arroyo What was the scenario we're trying to unblock with this?

Comment (jo-arroyo) [reply lib/msal-browser/src/request/SilentRequest.ts:50]: @tnorling This is being used in the benchmark tests. When we last talked about this, I was told not being able to pass this in SilentRequest was an oversight?

**Comment by jo-arroyo** [comment]
Workaround found offline. Closing as no longer needed.

**Pull request closed without merging**